File: PDOO/ruby p1/SuppliesPackage.rb
module Deepspace
    class SuppliesPackage
        def initialize(ap,fu,sp)
            @ammoPower=ap
            @fuelUnits=fu
            @shieldPower=sp
        end

        def self.newCopy(copy)
            new(copy.ammoPower,copy.fuelUnits, copy.shieldPower)
        end

        def ammoPower
            @ammoPower
        end

        def fuelUnits
            @fuelUnits
        end

        def shieldPower
            @shieldPower
        end
    end
end

File: PDOO/java p1/src/deepspace/SuppliesPackage.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author carmenc
 */
class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammopower, float fuelunits, float shieldpower){
        ammoPower = ammopower;
        fuelUnits = fuelunits;
        shieldPower = shieldpower;
    }
    
    SuppliesPackage(SuppliesPackage s) {
        //no se comparten direcciones de memoria
        ammoPower = s.getAmmoPower();
        fuelUnits = s.getFuelUnits();
        shieldPower = s.getShieldPower();
    }
    
    public float getAmmoPower(){
        return ammoPower;
    }
    
    public float getFuelUnits(){
        return fuelUnits;
    }
    
    public float getShieldPower(){
        return shieldPower;
    }
}